### PR TITLE
Refactor trigger handling

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: core/pkg/worker/instrumented_worker.go [hunk 1]
```diff
@@ -38,59 +39,56 @@ func InstrumentedWorkerFn(w Worker) Worker {
...
+func (a *instrumentedWorker) tracerFromTrigger(ctx context.Context, trigger *v1.TriggerRequest) (context.Context, trace.Span, error) {
+	if http := trigger.GetHttp(); http != nil {
+		ctx, s := otel.Tracer("membrane/pkg/worker", trace.WithInstrumentationVersion(span.MembraneVersion)).
+			Start(ctx, span.Name(http.Path))
+
+		s.SetAttributes(
+			semconv.CodeFunctionKey.String("HandleHttp"),
+			semconv.HTTPMethodKey.String(http.Method),
+			semconv.HTTPTargetKey.String(http.Path),
+			// semconv.HTTPURLKey.String(http.URL),
```
Comment (tjholm) [inline core/pkg/worker/instrumented_worker.go:51]: TODO: Replace this

File: core/pkg/worker/instrumented_worker.go [hunk 2]
```diff
@@ -38,59 +39,56 @@ func InstrumentedWorkerFn(w Worker) Worker {
...
+		return ctx, s, nil
+	} else if topic := trigger.GetTopic(); topic != nil {
+		ctx, s := otel.Tracer("membrane/pkg/worker", trace.WithInstrumentationVersion(span.MembraneVersion)).
+			Start(ctx, span.Name("topic-"+topic.Topic))
+
+		s.SetAttributes(
+			semconv.CodeFunctionKey.String("HandleEvent"),
+			semconv.MessagingDestinationKindTopic,
+			semconv.MessagingDestinationKey.String(topic.Topic),
+			// semconv.MessagingMessageIDKey.String(topic.),
```
Comment (tjholm) [inline core/pkg/worker/instrumented_worker.go:63]: TODO: Replace this

File: cloud/aws/runtime/gateway/lambda.go [hunk 3]
```diff
@@ -211,65 +213,64 @@ func (s *LambdaGateway) handle(ctx context.Context, data map[string]interface{})
...
+
+			wrkr, err := s.pool.GetWorker(&pool.GetWorkerOptions{
+				Trigger: request,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("unable to get worker to event trigger")
+			}
+
+			// FIXME: Restore this with contract update for topic/event context
+			// var mc propagation.MapCarrier = event.Attributes
```
Comment (tjholm) [inline cloud/aws/runtime/gateway/lambda.go:270]: TODO: Restore this

File: core/pkg/worker/instrumented_worker.go (hunk 2 above)
Comment (tjholm) [reply core/pkg/worker/instrumented_worker.go]: Not required so leaving out for now. Have added missing required key for system instead

**Review by asalkeld — APPROVED** [review]

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**